### PR TITLE
[gtk] Bump to gtk+-3.24.10 which is the first version using meson build

### LINKED ISSUE #6554: gtk missing resources
My gtk application ported to Windows via vcpkg crashes because it misses an essential resource ("image-missing.png"). As I found, that file is contained beneath gtk/icons and referenced in the file gtk/gtk.gresource.xml - which to my understanding should be compiled via glib-compile-resources and linked into the library. The CMakeLists.txt contained in vcpkg/ports/gtk does not mention any of this, so I assume it is simply not yet implemented.

### LINKED ISSUE #7412: [gtk] update to 3.22.30
Library name: GTK

New version number: v3.22.30

The current GTK build is very VERY old: v3.22.19 (circa 2017-08-22). I'm running into bugs that have been fixed a long time ago when using it. So I don't know if v3.22.30 is the right call, but it's the latest. I do know that  a 2 year old version is not OK. Could we update to something from this year at least?

I'm currently experimenting with just changing the portfile to point to the latest version and rebuilding. But I'm worried that other dependent libs are also very out of date. GTK has a long list of deps.

### LINKED ISSUE #7641: [gtk] missing `.pc` file for pkg-config
****
When I tried to find gtk in CMake like this, it failed: 
```cmake
find_package(PkgConfig REQUIRED)
pkg_check_modules(GTK3 REQUIRED gtk+-3.0)
```
There is no `.pc` fileassociated with GTK+ 3  in under `C:\vcpkg\installed\x64-windows\lib\pkgconfig`. 

**Proposed solution**
Add the `.pc` associated with GTK+ 3.

### LINKED ISSUE #9035: fribidi build failure on Linux
**Host Environment**
- OS: Linux (WSL)
- Compiler: gcc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0

**To Reproduce**
`./vcpkg install fribidi`

**Failure logs** 
```
the following packages will be built and installed:
    fribidi[core]:x64-linux
Starting package 1/1: fribidi:x64-linux
Building package fribidi[core]:x64-linux...
-- Using cached /home/ghasan/vcpkg/downloads/fribidi-fribidi-58c6cb390a9a18c98b2cbaac555d8ea9352a9e4f.tar.gz
-- Using source at /home/ghasan/vcpkg/buildtrees/fribidi/src/a9352a9e4f-1a97bf5d69
-- Configuring x64-linux-rel
-- Configuring x64-linux-rel done
-- Configuring x64-linux-dbg
-- Configuring x64-linux-dbg done
-- Package x64-linux-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: /usr/bin/ninja install -v
    Working Directory: /home/ghasan/vcpkg/buildtrees/fribidi/x64-linux-rel
    Error code: 1
    See logs for more information:
      /home/ghasan/vcpkg/buildtrees/fribidi/package-x64-linux-rel-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_meson.cmake:8 (vcpkg_execute_required_process)
  ports/fribidi/portfile.cmake:19 (vcpkg_install_meson)
  scripts/ports.cmake:94 (include)


Error: Building package fribidi:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: fribidi:x64-linux
  Vcpkg version: 2019.09.12-unknownhash
```

```
cat /home/ghasan/vcpkg/buildtrees/fribidi/package-x64-linux-rel-out.log
[1/63] cc  -Igen.tab/gen-unicode-version@exe -Igen.tab -I../src/a9352a9e4f-1a97bf5d69/gen.tab -Ilib -I../src/a9352a9e4f-1a97bf5d69/lib -I. -I../src/a9352a9e4f-1a97bf5d69/ -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 /DWIN32 /D_WINDOWS /W3 /utf-8 /MD /O2 /Gy /DNDEBUG /Z7 -UHAVE_CONFIG_H -DHAVE_STRINGIZE -DDONT_HAVE_FRIBIDI_CONFIG_H -DHAVE_STDLIB_H -DHAVE_STRING_H -DHAVE_STRINGS_H -DSTDC_HEADERS=1 -MD -MQ 'gen.tab/gen-unicode-version@exe/gen-unicode-version.c.o' -MF 'gen.tab/gen-unicode-version@exe/gen-unicode-version.c.o.d' -o 'gen.tab/gen-unicode-version@exe/gen-unicode-version.c.o' -c ../src/a9352a9e4f-1a97bf5d69/gen.tab/gen-unicode-version.c
FAILED: gen.tab/gen-unicode-version@exe/gen-unicode-version.c.o
cc  -Igen.tab/gen-unicode-version@exe -Igen.tab -I../src/a9352a9e4f-1a97bf5d69/gen.tab -Ilib -I../src/a9352a9e4f-1a97bf5d69/lib -I. -I../src/a9352a9e4f-1a97bf5d69/ -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 /DWIN32 /D_WINDOWS /W3 /utf-8 /MD /O2 /Gy /DNDEBUG /Z7 -UHAVE_CONFIG_H -DHAVE_STRINGIZE -DDONT_HAVE_FRIBIDI_CONFIG_H -DHAVE_STDLIB_H -DHAVE_STRING_H -DHAVE_STRINGS_H -DSTDC_HEADERS=1 -MD -MQ 'gen.tab/gen-unicode-version@exe/gen-unicode-version.c.o' -MF 'gen.tab/gen-unicode-version@exe/gen-unicode-version.c.o.d' -o 'gen.tab/gen-unicode-version@exe/gen-unicode-version.c.o' -c ../src/a9352a9e4f-1a97bf5d69/gen.tab/gen-unicode-version.c
cc: error: /DWIN32: No such file or directory
cc: error: /D_WINDOWS: No such file or directory
cc: error: /W3: No such file or directory
cc: error: /utf-8: No such file or directory
cc: error: /MD: No such file or directory
cc: error: /O2: No such file or directory
cc: error: /Gy: No such file or directory
cc: error: /DNDEBUG: No such file or directory
cc: error: /Z7: No such file or directory
ninja: build stopped: subcommand failed.
```

**Additional context**
Windows x86 and x64 builds like a breeze.

### PR DESCRIPTION
**Pull request by fungos**
**WORK-IN-PROGRESS - I still need to do more tests on Windows.**
**If your issue is mentioned here, please help review and test this PR.**

- Fixes #7412 by using gtk 3.24.10.

- Possibly fixes #6554, #7827, #8103 as now all the missing assets are
correctly copied into installed/<triplet>/share/ and should be deployed
with the final application.

This simplify the gtk port and correctly generate the missing files.

The source of the problem is with the underlying build system, as these
files were installed by the demo `icon-browser`, but the
gtk-update-icon-cache were executed by the main gtk project, failing if
the demo wasn't built or installed.

- This also fixes #7641 as now the meson build will correctly generate
the missing .pc files.

- This also fixes #9035, #7150 by fixing `vcpkg_configure_meson.cmake`
which fails on Linux/OSX because it indiscriminately try to pass msvc
cl arguments to the cmake c/xx flags. Although this works, it is not
the ideal fix. The correct fix would be to do as suggested in
issue #8271. Also see issue #8639.

### CODE REVIEW

**Comment by cenit** [comment]
I was also working on improving a lot meson/pkgconfig here on vcpkg. I might already import your commit in my PR to fix gtk (one of the many libraries that are not in a very good shape here when built in *nix)

**Comment by fungos** [comment]
Need to hold on this as Meson is basically useless on windows. I need do some hacks to get this to  find all its dependencies as meson tries pkg_config and some other cmake instance elsewhere, which in turns fail to find the right paths.

**Comment by Illation** [comment]
Is it possible that this also fixes #3339 ?

**Comment by fungos** [comment]
Unfortunately no. But looking there, the fix referenced in there is probably your best bet. I suggest taking that and doing a PR since the person don't want to.
Also, if some recent Gtkmm uses Mason, if I find a way to get this working, then fixing Gtkmm should be something similar, but not this.

File: ports/gtk/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: gtk
-Version: 3.22.19-3
+Version: 3.24.10-1
 Homepage: https://www.gtk.org/
 Description: Portable library for creating graphical user interfaces.
-Build-Depends: glib, atk, gdk-pixbuf, pango, cairo, libepoxy, gettext
+Build-Depends: tool-meson, glib, atk, gdk-pixbuf, pango, cairo, gettext, libepoxy
```
Comment (windelbouwman) [inline ports/gtk/CONTROL]: Why were the last two items swapped?

File: ports/gtk/cmake/install_headers.cmake [hunk 2]
```diff
@@ -1,376 +0,0 @@
-install(FILES gdk/win32/gdkwin32.h DESTINATION include/gdk)
-install(FILES gdk/gdk.h DESTINATION include/gdk)
-install(FILES gdk/gdk-autocleanup.h DESTINATION include/gdk)
-install(FILES gdk/gdkapplaunchcontext.h DESTINATION include/gdk)
```
Comment (windelbouwman) [inline ports/gtk/cmake/install_headers.cmake]: Having this file dropped seems like a real good action!

File: ports/gtk/portfile.cmake [hunk 3]
```diff
@@ -1,42 +1,34 @@
...
 
-file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})
-file(COPY ${CMAKE_CURRENT_LIST_DIR}/cmake DESTINATION ${SOURCE_PATH})
+vcpkg_configure_meson(
+	SOURCE_PATH ${SOURCE_PATH}
+	OPTIONS
+		-Dexamples=false
+		-Dtests=false
+		--backend=ninja
+		#-Ddemos=false # WE NEED demos, because the icons come from here, otherwise the update icon cache will fail to build!
```
Comment (windelbouwman) [inline ports/gtk/portfile.cmake]: As a pragmatic approach I suggest to leave demos = true, and put a TODO in this code with a reference to an issue in the GTK sourcecode which addresses this bug in the upstream package.

We will then have a solution which works.

**Comment by JackBoosY** [comment]
@fungos  Any progress in this PR?

**Comment by fungos** [comment]
Unfortunately, I do not have anymore interest on GTK at all due their own lack of interest in improving their build system.

**Pull request closed without merging**